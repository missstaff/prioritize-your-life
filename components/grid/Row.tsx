import { ScaledSheet, s, vs } from "react-native-size-matters";
import AppThemedView from "../app_components/AppThemedView";

interface RowProps {
  children: React.ReactNode;
}

const Row = ({ children, ...props }: RowProps) => {
  return (
    <AppThemedView style={styles.row} {...props}>
      {children}
    </AppThemedView>
  );
};

const styles = ScaledSheet.create({
  row: {
    alignItems: "center",
    display: "flex",
    flexDirection: "row",
    justifyContent: "flex-start",
    paddingHorizontal: s(5),
    paddingVertical: vs(5),
    width: "100%",
  },
});

export default Row;
