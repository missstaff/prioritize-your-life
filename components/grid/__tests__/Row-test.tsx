import React from "react";
import { Text } from "react-native";
import { render } from "@testing-library/react-native";
import Row from "../Row";

describe("Row Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly with no children", () => {
    const { toJSON } = render(<Row children={undefined} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with children", () => {
    const { getByText } = render(
      <Row>
        <Text>Test Child</Text>
      </Row>
    );
    expect(getByText("Test Child")).toBeTruthy();
  });
});
