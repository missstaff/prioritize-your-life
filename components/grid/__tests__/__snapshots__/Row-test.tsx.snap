// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component renders correctly with no children 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#151718",
      },
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingHorizontal": 10.714285714285714,
        "paddingVertical": 9.808823529411764,
        "width": "100%",
      },
    ]
  }
  testID="row"
/>
`;
